src/console_gui.cpp

```diff
@@ -198,8 +142,13 @@ struct IConsoleWindow : Window
 	 */
 	void Scroll(int amount)
 	{
-		int max_scroll = std::max(0, IConsoleLine::size + 1 - this->height / this->line_height);
-		IConsoleWindow::scroll = Clamp<int>(IConsoleWindow::scroll + amount, 0, max_scroll);
+		size_t max_scroll = std::max<size_t>(0, _iconsole_buffer.size() + 1 - this->height / this->line_height);
```

Comment:
`size_t` is unsigned. This clamping to zero won't work, but create huge numbers.